packages/textfield/src/mwc-textfield-base.ts

```diff
@@ -287,7 +287,7 @@ export abstract class TextFieldBase extends FormElement {
     }
 
     if (changedProperties.has('value') &&
-        changedProperties.get('value') !== undefined) {
```

Comment:
This was added to fix https://github.com/material-components/material-components-web-components/issues/515. I'm not sure what this change is hoping to achieve by removing the functionality.

This goes from "if the value _was previously_ not undefined" to "if the value _is currently_ not undefined", which isn't what the issue was solving.

Reply:
If I understand correctly, the fix was about to call Foundation `setValue`  when value changes. This does not change. 

#515 and #1116 have a similar root cause which is the component's inability to re-evaluate positioning logic when value change. 

Then problem of `changedProperties.get('value') !== undefined)` is that when you have an `undefined` value at some stage, the positioning logic fail again to be called  next time the value changes (see https://codepen.io/christophe-g/pen/ZEbGMwp)